### PR TITLE
Adding Creds UI Wrapper

### PR DESCRIPTION
**Pull request by jayasimha-raghavan-unskript**
## Description
This PR brings Creds App (Wrapper around creds-ui) that is a standalone app to
create credential that conform to our Credential-Save Schema.

### Testing
Please describe the tests that you ran to verify your changes. Please summarize what did you test and what needs to be tested e.g. deployed and tested helm chart locally. 

https://github.com/unskript/Awesome-CloudOps-Automation/assets/87547684/e604b35c-4e4c-4259-ad12-d0f9018a2cc4

#### Extending Docker

https://github.com/unskript/Awesome-CloudOps-Automation/assets/87547684/b161b594-bbd5-4729-8ce0-235ade565891

### Checklist:
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] Any dependent changes have been merged and published. 

### Documentation
Make sure that you have documented corresponding changes in this repository.

### CODE REVIEW

File: build/templates/Dockerfile.template [hunk 1]
```diff
@@ -1,6 +1,8 @@
 FROM unskript/awesome-runbooks:latest as base 
-COPY custom/actions /unskript/data
-COPY custom/runbooks /unskript/data
+COPY custom/actions/. /unskript/data/actions/
+COPY custom/runbooks/. /unskript/data/runbooks/
 
-CMD ["./start.sh"]
+# The creds directory is the directory you mentioned while running the creds_app.py with the -o option
+COPY <CREDS_DIRECTORY>/. /unskript/credential/.local/share/jupyter/metadata/credential-save/
```
Comment (amit-chandak-unskript) [inline build/templates/Dockerfile.template:6]: Have we tested this?  with a docker build?

File: unskript-ctl/creds_app.py [hunk 2]
```diff
@@ -0,0 +1,946 @@
...
+
+def main():
+    """main: This is the Main function that interfaces with the creds-ui"""
+    try:
+        schema_json = json.loads(credential_schemas)
+    except Exception as e:
+        print(f"Exception occured {e}")
+        return
+  
+    parser = ArgumentParser(prog='unskript-ctl')
```
Comment (amit-chandak-unskript) [inline unskript-ctl/creds_app.py:911]: this is not unskript-ctl

File: unskript-ctl/creds_ui.py [hunk 3]
```diff
@@ -53,7 +53,10 @@
 
 # This variable is used to hold the Credential directory
 # Where all the creds are saved
-CREDS_DIR = os.environ.get('HOME') + "/.local/share/jupyter/metadata/credential-save/"
+if os.environ.get('CREDS_DIR') != None:
```
Comment (amit-chandak-unskript) [inline unskript-ctl/creds_ui.py]: please fix the pylint here, github is showing something

File: unskript-ctl/creds_ui.py [hunk 4]
```diff
@@ -53,7 +53,10 @@
...
+    CREDS_DIR = os.environ.get('CREDS_DIR')
```
Comment (amit-chandak-unskript) [inline unskript-ctl/creds_ui.py:57]: What is the change in this file?

File: unskript-ctl/creds_app.py [hunk 5]
```diff
@@ -0,0 +1,946 @@
...
+    "required": [
+      "Username",
+      "Password",
+      "Security_Token"
+    ]
+  }
+  ]
+'''
+
+def create_stub_cred_files(dirname: str):
```
Comment (amit-chandak-unskript) [inline unskript-ctl/creds_app.py:882]: why is this needed?

File: unskript-ctl/stub_creds.json [hunk 6]
```diff
@@ -0,0 +1,217 @@
+[
```
Comment (amit-chandak-unskript) [inline unskript-ctl/stub_creds.json:1]: Why do we need this file

File: unskript-ctl/creds_ui.py [hunk 7]
```diff
@@ -53,7 +53,10 @@
 
 # This variable is used to hold the Credential directory
 # Where all the creds are saved
-CREDS_DIR = os.environ.get('HOME') + "/.local/share/jupyter/metadata/credential-save/"
+if os.environ.get('CREDS_DIR') is not None:
```
Comment (amit-chandak-unskript) [inline unskript-ctl/creds_ui.py:56]: This needs some detailed comment as to whats happening here

File: build/templates/Dockerfile.template (hunk 1 above)
Comment (jayasimha-raghavan-unskript) [reply build/templates/Dockerfile.template]: Yes, I have attached Video to the PR

File: unskript-ctl/creds_app.py (hunk 2 above)
Comment (jayasimha-raghavan-unskript) [reply unskript-ctl/creds_app.py]: Done

File: unskript-ctl/creds_ui.py (hunk 3 above)
Comment (jayasimha-raghavan-unskript) [reply unskript-ctl/creds_ui.py]: Done

**Review by amit-chandak-unskript — APPROVED** [review]
